src/app/zap-templates/templates/app/helper.js

```diff
@@ -223,8 +252,33 @@ function chip_endpoint_cluster_list()
     } else {
       mask = c.mask.map((m) => `ZAP_CLUSTER_MASK(${m.toUpperCase()})`).join(' | ')
     }
-    ret = ret.concat(`  { ${c.clusterId}, ZAP_ATTRIBUTE_INDEX(${c.attributeIndex}), ${c.attributeCount}, ${c.attributeSize}, ${
-        mask}, ${functionArray} }, /* ${c.comment} */ \\\n`)
+
+    clientGeneratedCommands = [];
+    serverGeneratedCommands = [];
+    c.commands.forEach((cmd) => {
+      console.log(cmd);
+      if (cmd.mask.includes('incoming_server')) {
+        clientGeneratedCommands.push(`${cmd.commandId} /* ${cmd.name} */`);
+      }
+      if (cmd.mask.includes('incoming_client')) {
+        serverGeneratedCommands.push(`${cmd.commandId} /* ${cmd.name} */`);
+      }
+    });
+    clientGeneratedCommands.push("0xFFFF'FFFF /* end of list */");
+    serverGeneratedCommands.push("0xFFFF'FFFF /* end of list */");
+
+    ret = ret.concat(`  { \\
+      .clusterId = ${c.clusterId},  \\
+        .attributes = ZAP_ATTRIBUTE_INDEX(${c.attributeIndex}), \\
+        .attributeCount = ${c.attributeCount}, \\
+        .clusterSize = ${c.attributeSize}, \\
+        .mask = ${mask}, \\
+        .functions = ${functionArray}, \\
+        .clientGeneratedCommandList = ZAP_GENERATED_COMMANDS_INDEX( ${totalCommands} ), \\
+        .serverGeneratedCommandList = ZAP_GENERATED_COMMANDS_INDEX( ${totalCommands + clientGeneratedCommands.length} ) \\
+    }, /* ${c.comment} */ \\\n`)
```

Comment:
With this change to use field names, the comment really needs to go before the struct init; otherwise it looks very very confusing.

Reply:
The same, will add // clangformat off to disable formatter here.

Reply:
It's not just the formatter, it's the placement.

Reply:
The same, will add // clangformat off to disable formatter here.

Reply:
Template updated